src/toil/batchSystems/kubernetes.py

```diff
@@ -496,47 +492,126 @@ def _getUpdatedBatchJobImmediately(self):
+                jobObject = j
+                chosenFor = 'done'
+
+            if jobObject is None:
+                for j in self._ourJobObjects():
+                    # If there aren't any succeeded jobs, scan all jobs
+                    # See how many times each failed
+                    failCount = getattr(j.status, 'failed', 0)
+                    if failCount is None:
+                        # Make sure it is an int
+                        failCount = 0
+                    if failCount > 0:
+                        # Take the first failed one you find
+                        jobObject = j
+                        chosenFor = 'failed'
+                        break
+
+            if jobObject is None:
+                # If no jobs are failed, look for jobs with pods with
+                # containers stuck in Waiting with reason ImagePullBackOff
+                for j in self._ourJobObjects():
+                    pod = self._getPodForJob(j)
+
+                    if pod is None:
+                        # Skip jobs with no pod
+                        continue
+
+                    waitingInfo = getattr(pod.status.container_statuses[0].state, 'waiting')
+                    if waitingInfo is not None and waitingInfo.reason == 'ImagePullBackOff':
+                        # Assume it will never finish, even if the registry comes back or whatever.
+                        # We can get into this state when we send in a non-existent image.
+                        # See https://github.com/kubernetes/kubernetes/issues/58384
+                        jobObject = j
+                        chosenFor = 'stuck'
+                        logger.warning('Failing stuck job; did you try to run a non-existent Docker image?'
+                                       ' Check TOIL_APPLIANCE_SELF.')
+                        break
+           
+            w = kubernetes.watch.Watch()    
+            if jobObject is None:
+                # Nothing is ready yet.
+
+                if self.enableWatching:
+                    for j in self._ourJobObjects():
+                        print(j.spec.template.metadata.labels[u'job-name'], type(j.spec.template.metadata.labels[u'job-name']))
+                        for event in w.stream(self.coreApi.list_namespaced_pod, self.namespace, timeout_seconds=maxWait):
+                            job = event['object']
+                            if job.metadata.name.startswith(self.jobPrefix):
+                                logger.info("Event: %s %s %s" % (event['type'],event['object'].kind, event['object'].metadata.name))
+                                if job.status.phase == 'Failure':
+                                    logger.info("FAILED")
+                                    logger.warning(job.status.container_status[0].state.reason,
+                                        job.status.container_statuses[0].state.terminated.exit_code)
+                                    return None
```

Comment:
The method is supposed to tell the caller about failed jobs; they do in fact count as "updated". Here if the job fails you're just returning None, which isn't right.